src/main/java/com/google/api/codegen/gapic/MainGapicProviderFactory.java

```diff
@@ -142,6 +149,24 @@
 
       return Arrays.<GapicProvider<? extends Object>>asList(mainProvider, packageInfoProvider);
 
+    } else if (id.equals(JAVA_SURFACE)) {
+      GapicCodePathMapper javaPathMapper =
+          CommonGapicCodePathMapper.newBuilder()
+              .setPrefix("src/main/java")
+              .setShouldAppendPackage(true)
+              .build();
+      GapicProvider<? extends Object> mainProvider =
+          SurfaceGapicProvider.newBuilder()
+              .setModel(model)
+              .setSnippetSetRunner(
+                  new CommonSnippetSetRunner(
+                      SnippetSetRunner.SNIPPET_RESOURCE_ROOT, new JavaSnippetUtil()))
```

Comment:
Hmm yeah it doesn't make sense to pass the resource root in here any more. Maybe it could be provided by ViewModel instead.